apps/ausweis/src/use-cases/ReceivePidUseCaseBPrimeFlow.ts

```diff
@@ -210,36 +211,44 @@ export class ReceivePidUseCaseBPrimeFlow {
 
       const authorizationCodeResponse = await fetch(this.refreshUrl)
       if (!authorizationCodeResponse.ok) {
-        this.handleError()
+        this.handleError(`Auth code response has invalid status. ${authorizationCodeResponse.status}`)
         return
       }
 
-      const authorizationCode = new URL(authorizationCodeResponse.url).searchParams.get('code')
+      const { pin_nonce: pinNonce } = await authorizationCodeResponse.json()
 
-      if (!authorizationCode) {
-        this.handleError()
-        return
-      }
+      const pinDerivedEph = await deriveKeypairFromPin(this.options.agent.context, this.options.pidPin)
+
+      const code = await requestToPidProvider(
+        authorizationCodeResponse.url,
+        this.options.agent as unknown as FullAppAgent,
+        pinDerivedEph,
+        pinNonce
+      )
 
       this.accessToken = await acquireAccessToken({
         resolvedCredentialOffer: this.resolvedCredentialOffer,
         resolvedAuthorizationRequest: {
           ...this.resolvedAuthorizationRequest,
-          code: authorizationCode,
+          code,
         },
         agent: this.options.agent,
+        // @ts-ignore
```

Comment:
why ts-ignore? acquireAccesstoken is from internal agent package right, so we should be able to type this correctly?